raft.go

```diff
@@ -881,6 +881,8 @@ func (r *Raft) checkLeaderLease() time.Duration {
 				} else {
 					r.logger.Debug("failed to contact", "server-id", server.ID, "time", diff)
 				}
+
+				r.observe(FailedHeartbeatObservation{PeerID: server.ID})
```

Comment:
What do you think about adding the LastContact time for this peer here? That way decisions can be made based on how long this node has been unreachable.

Reply:
That's a good addition, I'll put that in